src/sources/aws_kinesis_firehose/handlers.rs

```diff
@@ -71,18 +80,52 @@ pub(super) async fn firehose(
                         })
                         .unwrap_or((None, None));
 
+                    let now = Utc::now();
                     for event in &mut events {
                         if let Some(batch) = &batch {
                             event.add_batch_notifier(batch.clone());
                         }
                         if let Event::Log(ref mut log) = event {
-                            log.try_insert(
+                            log_namespace.insert_vector_metadata(
+                                log,
                                 log_schema().source_type_key(),
-                                Bytes::from("aws_kinesis_firehose"),
+                                path!("source_type"),
+                                Bytes::from_static(AwsKinesisFirehoseConfig::NAME.as_bytes()),
+                            );
+                            // This handles the transition from the original timestamp logic. Originally the
+                            // `timestamp_key` was always populated by the `request.timestamp` time.
+                            match log_namespace {
+                                LogNamespace::Vector => {
+                                    log.metadata_mut()
+                                        .value_mut()
+                                        .insert(path!("vector", "ingest_timestamp"), now);
```

Comment:
This works fine, but you could switch it to `log.insert(metadata_path!("vector", "ingest_timestamp"))`
which I think is slightly easier to read.

Reply:
Resolved in https://github.com/vectordotdev/vector/pull/15189/commits/74dd2607f0a2beaf8546118fbbda1c919a326d17, and updated the other places I had used the same pattern